spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/BasicJdbcConverter.java

```diff
@@ -70,6 +76,34 @@ public BasicJdbcConverter(
 		super(context, conversions);
 	}
 
+	/**
+	 * Only provided as a bridge to the old way of passing additional parameters to an insert statement.
+	 *
+	 * @param additionalParameters
+	 */
+	public static Identifier fromNamedValues(Map<String, Object> additionalParameters) {
+
+		List<Identifier.SingleIdentifierValue> keys = new ArrayList<>();
+		additionalParameters.forEach((k, v) -> keys.add(new Identifier.SingleIdentifierValue(k, v, v == null ? Object.class : v.getClass())));
```

Comment:
By using factory/mutator methods on `Identifier`, there should not be a need to instantiate `SingleIdentifierValue` from within the code.